recipe/meta.yaml

```diff
@@ -8,31 +8,46 @@ source:
   url: https://github.com/Unidata/netcdf-c/archive/v{{ version }}.tar.gz
   sha256: a2fabf27c72a5ee746e3843e1debbaad37cd035767eaede2045371322211eebb
   patches:
-    - CMakeLists.patch  # [win]
+    # These patches are needed but there is no way I can get them to work on AppVeyor's Windows image :-(
+    - 0001-Fix-bug-Index-exceeds-dimension-bound.patch  # [not win]
+    - 0002-run_get_hdf4_files.sh-Use-curl-if-wget-fails-macOS-d.patch  # [not win]
+    - 0003-Add-find_package-Threads-REQUIRED-to-CMakeLists.txt.patch  # [not win]
+    - 0004-Prefer-getenv-TOPSRCDIR-over-STRINGIFY-TOPSRCDIR.patch  # [not win]
+    - 0005-Convert-some-c99-to-c89-by-hand.patch  # [not win]
+    - 0006-Fix-tests-for-Windows-VS2008-define-__func__-as-__FU.patch  # [not win]
+    - 0007-Fix-broken-test.patch  # [not win]
+    - 0008-Finish-the-missing-code-to-handle-VS-in-test_common..patch  # [not win]
 
 build:
-  number: 3
-  skip: True  # [win and vc<14]
+  number: 4
   run_exports:
     #   https://abi-laboratory.pro/tracker/timeline/netcdf/
     - {{ pin_subpackage('libnetcdf') }}
 
 requirements:
   build:
     - cmake
+    - make  # [not win]
     - pkg-config  # [not win]
+    # Needed for tests: Bash shell not found; disabling shell script tests.
+    - posix  # [win]
+    - m2-m4  # [win]
```

Comment:
@ocefpaf I found that `posix` and `m2-m4` seemed to break the build for me (invalid escape character messages from `cmake`). YMMV. I didn't get the invalid line endings, but this may be related.
Since the tests are skipped on Windows maybe we can remove this for now?

I had to add the following patch to prevent the MS runtime libs being installed again:
```
--- CMakeLists.txt.orig	2018-08-02 17:48:12.942762500 +1000
+++ CMakeLists.txt	2018-08-02 17:48:15.731124100 +1000
@@ -1713,7 +1713,7 @@
 # Subdirectory CMakeLists.txt files should specify their own
 # 'install' files.
 # Including 'CPack' kicks everything off.
-INCLUDE(InstallRequiredSystemLibraries)
+#INCLUDE(InstallRequiredSystemLibraries)
 CONFIGURE_FILE(
   ${CMAKE_CURRENT_SOURCE_DIR}/FixBundle.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/FixBundle.cmake
```